tests/functional/placementapi_controller_test.go

```diff
@@ -170,4 +171,145 @@ var _ = Describe("PlacementAPI controller", func() {
 				ContainSubstring("www_authenticate_uri = %s", keystoneAPI.Status.APIEndpoints["public"]))
 		})
 	})
+
+	When("A PlacementAPI is created with service override", func() {
+		BeforeEach(func() {
+			DeferCleanup(k8sClient.Delete, ctx, CreatePlacementAPISecret(namespace, SecretName))
+			DeferCleanup(th.DeleteKeystoneAPI, th.CreateKeystoneAPI(placementApiName.Namespace))
+
+			spec := GetDefaultPlacementAPISpec()
+			serviceOverride := map[string]interface{}{}
+			serviceOverride["internal"] = map[string]interface{}{
+				"metadata": map[string]map[string]string{
+					"annotations": {
+						"dnsmasq.network.openstack.org/hostname": "placement-internal.openstack.svc",
+						"metallb.universe.tf/address-pool":       "osp-internalapi",
+						"metallb.universe.tf/allow-shared-ip":    "osp-internalapi",
+						"metallb.universe.tf/loadBalancerIPs":    "internal-lb-ip-1,internal-lb-ip-2",
+					},
+					"labels": {
+						"internal": "true",
+						"service":  "placement",
+					},
+				},
+				"spec": map[string]interface{}{
+					"type": "LoadBalancer",
+				},
+			}
+
+			spec["override"] = map[string]interface{}{
+				"service": serviceOverride,
+			}
+
+			placementAPI := CreatePlacementAPI(placementApiName, spec)
+			DeferCleanup(
+				th.DeleteDBService,
+				th.CreateDBService(
+					placementApiName.Namespace,
+					GetPlacementAPI(placementApiName).Spec.DatabaseInstance,
+					corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{{Port: 3306}},
+					},
+				),
+			)
+
+			th.SimulateMariaDBDatabaseCompleted(placementApiName)
+			th.SimulateJobSuccess(types.NamespacedName{
+				Namespace: placementApiName.Namespace,
+				Name:      fmt.Sprintf("%s-db-sync", placementApiName.Name),
+			})
+			th.SimulateDeploymentReplicaReady(placementApiName)
+			th.SimulateKeystoneServiceReady(placementApiName)
+			th.SimulateKeystoneEndpointReady(placementApiName)
+			DeferCleanup(th.DeleteInstance, placementAPI)
+		})
+
+		It("creates KeystoneEndpoint", func() {
+			keystoneEndpoint := th.GetKeystoneEndpoint(placementApiName)
+			endpoints := keystoneEndpoint.Spec.Endpoints
+			Expect(endpoints).To(HaveKeyWithValue("public", "http://placement-public."+placementApiName.Namespace+".svc:8778"))
```

Comment:
do we want to also test for api versions locators?

Reply:
```
$ openstack endpoint list | grep placement
| a15f64a122184e4e83e761e29554e56f | regionOne | placement    | placement    | True    | internal  | http://placement-internal.openstack.svc:8778                         |
| ca6099e9709f40e49331fa37aae014f1 | regionOne | placement    | placement    | True    | public    | http://placement-public-openstack.apps-crc.testing                   |
```
They are not used in case of placement